code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -13,65 +11,68 @@
 	max_w_class = ITEMSIZE_NORMAL
 	max_storage_space = 14 //The sum of the w_classes of all the items in this storage item.
 	req_access = list(access_armory)
-	var/locked = 1
-	var/broken = 0
+	var/locked = TRUE
+	var/broken = FALSE
 	var/icon_locked = "lockbox+l"
 	var/icon_closed = "lockbox"
 	var/icon_broken = "lockbox+b"
 
-
-	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/card/id))
-			if(src.broken)
-				to_chat(user, "<span class='warning'>It appears to be broken.</span>")
+/obj/item/storage/lockbox/attackby(obj/item/I, mob/user)
+	var/obj/item/card/id/ID = I.GetID()
+	if(istype(ID))
+		if(broken)
+			to_chat(user, SPAN_WARNING("It appears to be broken."))
+			return
+		if(allowed(user))
+			locked = !locked
+			if(locked)
+				icon_state = icon_locked
+				to_chat(user, SPAN_NOTICE("You lock \the [src]!"))
 				return
-			if(src.allowed(user))
-				src.locked = !( src.locked )
-				if(src.locked)
-					src.icon_state = src.icon_locked
-					to_chat(user, "<span class='notice'>You lock \the [src]!</span>")
-					return
-				else
-					src.icon_state = src.icon_closed
-					to_chat(user, "<span class='notice'>You unlock \the [src]!</span>")
-					return
 			else
-				to_chat(user, "<span class='warning'>Access Denied</span>")
-		else if(istype(W, /obj/item/melee/energy/blade))
-			if(emag_act(INFINITY, user, W, "The locker has been sliced open by [user] with an energy blade!", "You hear metal being sliced and sparks flying."))
-				var/obj/item/melee/energy/blade/blade = W
-				blade.spark_system.queue()
-				playsound(src.loc, 'sound/weapons/blade.ogg', 50, 1)
-				playsound(src.loc, /decl/sound_category/spark_sound, 50, 1)
-		if(!locked)
-			..()
+				icon_state = icon_closed
+				to_chat(user, SPAN_NOTICE("You unlock \the [src]!"))
+				return
 		else
-			to_chat(user, "<span class='warning'>It's locked!</span>")
+			to_chat(user, SPAN_WARNING("Access denied."))
+			return
+
+	if(istype(I, /obj/item/melee/energy/blade))
+		if(emag_act(INFINITY, user, I, "The locker has been sliced open by [user] with an energy blade!", "You hear metal being sliced and sparks flying."))
+			var/obj/item/melee/energy/blade/blade = I
+			blade.spark_system.queue()
+			playsound(src.loc, 'sound/weapons/blade.ogg', 50, 1)
+			playsound(src.loc, /decl/sound_category/spark_sound, 50, 1)
+			return
+
+	if(locked)
```

Comment:
Maybe the locked message shouldn't appear if you've just hit it with an energy blade / anything that isn't a hand or an ID?

Getting:
You hear metal being sliced and sparks flying.
It's locked!
You hear metal being sliced and sparks flying.
It's locked!
You hear metal being sliced and sparks flying.
It's locked!

Seems weird to me while you abuse the box with a lightsaber. Stick an else in there?

Reply:
![image](https://user-images.githubusercontent.com/22774890/124495323-436efc00-ddb8-11eb-88e3-d643397cb3c1.png)

it already has a return, and appears to work just fine

Reply:
Oops. You're quite right.